packages/gestalt/src/SheetMobile/Header.js

```diff
@@ -150,16 +150,28 @@ export default function Header({
         {primaryAction ? (
           // Allow button text to wrap on mobile
           <Flex.Item flex="shrink">
-            <PrimaryAction
-              accessibilityLabel={primaryAction.accessibilityLabel}
-              href={primaryAction.href}
-              rel={primaryAction?.rel}
-              target={primaryAction?.target}
-              label={primaryAction.label}
-              onClick={({ event }) =>
-                primaryAction?.onClick({ event, onDismissStart: onExternalDismiss })
-              }
-            />
+            {primaryAction.href ? (
```

Comment:
shoudn't this be 

`{primaryAction.role === 'link' 
`
for consistency

Reply:
Will do this as a follow-up on [GESTALT-6942](https://jira.pinadmin.com/browse/GESTALT-6942) to minimize the impact on this PR a bit.